app/src/main/java/com/droidhats/campuscompass/viewmodels/SearchViewModel.kt

```diff
@@ -0,0 +1,111 @@
+
+
+class SearchViewModel(application: Application) : AndroidViewModel(application) {
+
+    internal var googleSearchSuggestions = MutableLiveData<MutableList<GooglePlace>>()  //google places search results to be displayed
+    internal var indoorSearchSuggestions : LiveData<List<IndoorLocation>>? = null // concordia indoor location search results from SQLite database to be displayed
+    internal var searchSuggestions =  MutableLiveData<List<Location>>()  // The combined indoor and google search results to be displayed
+
+    private lateinit var indoorLocationDatabase: IndoorLocationDatabase
+    private lateinit var placesClient : PlacesClient // Used to query google places
+    private lateinit var indoorLocationRepository : IndoorLocationRepository //Used to query indoorLocations
+
+    private val context = getApplication<Application>().applicationContext
+
+    fun init(){
+        initPlacesSearch()
+        indoorLocationDatabase = Room.inMemoryDatabaseBuilder(context, IndoorLocationDatabase::class.java).build()
+        indoorLocationRepository = IndoorLocationRepository.getInstance(IndoorLocationDatabase.getInstance(context).indoorLocationDao())
+    }
+
+    private fun initPlacesSearch() {
+        Places.initialize(context, context.resources.getString(R.string.ApiKey), Locale.CANADA)
+        placesClient = Places.createClient(context)
+    }
+
+    private fun sendGooglePlacesQuery(query : String) : Boolean{
+        var success = false
+        //Set up your query here
+        val token : AutocompleteSessionToken = AutocompleteSessionToken.newInstance()
+        //Here you would bound your search (to montreal for example)
+        val bounds : RectangularBounds = RectangularBounds.newInstance(LatLng(45.011585, -73.812401), LatLng(46.166268, -73.514113))
+        val request : FindAutocompletePredictionsRequest = FindAutocompletePredictionsRequest.builder()
+            .setLocationBias(bounds)
+            .setSessionToken(token)
+            .setQuery(query)
+            .build()
+
+        //Get your query results here
+        val queryResults  = arrayListOf<GooglePlace>()
+        placesClient.findAutocompletePredictions(request).addOnSuccessListener {
+
+            for ( prediction in it.autocompletePredictions) {
+                queryResults.add(GooglePlace(prediction.placeId,
+                    prediction.getPrimaryText(null).toString(),
+                    prediction.getSecondaryText(null).toString(),
+                    LatLng(0.0, 0.0)))
+            }
+            if (queryResults.size > 0)
+                success = true
+
+            googleSearchSuggestions.value = queryResults
+
+        }.addOnFailureListener {
+            if (it is ApiException) {
+                val apiException =  it
+                Log.e(ContentValues.TAG, "Place not found: " + apiException.statusCode)
+            }
+        }
+        return success
+    }
+
+    private fun sendSQLiteQuery(query : String) : Boolean
```

Comment:
the formatting for the types on this line are weird, also the curly bracket should be on this line

Reply:
Formating and curly has been fixed 😄